### PR TITLE
[#13] [Part 1] Implement the base Clean Architecture with MVVM: data and domain layer

### LINKED ISSUE #13: Implement the base Clean Architecture with MVVM: data and domain layer
## Why

The template should provide the initial base architecture to start. The selected architecture for the shared, Android and iOS modules is [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) with MVVM.

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- #13 

- [x] Add `buildSrc` and extract dependencies
- [x] Add `Koin` DI
- [ ] Add `Ktor`

## What happened 👀

The template should provide the initial base architecture to start. The selected architecture for the shared, Android and iOS modules is [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) with MVVM.

## Insight 📝

N/A

## Proof Of Work 📹

Girl in a jacket

Screenshot 2023-08-10 at 3 48 52 PM

### CODE REVIEW

File: shared/shared.podspec [hunk 1]
```diff
@@ -35,5 +35,5 @@ Pod::Spec.new do |spec|
             SCRIPT
         }
     ]
-
-end
+                
+end
```
Comment (phongvhd93) [inline shared/shared.podspec]: missing blank line at EoF

File: shared/shared.podspec [hunk 2]
```diff
@@ -9,13 +9,13 @@ Pod::Spec.new do |spec|
     spec.vendored_frameworks      = 'build/cocoapods/framework/shared.framework'
     spec.libraries                = 'c++'
     spec.ios.deployment_target = '14.1'
-
-
+
```
Comment (phongvhd93) [inline shared/shared.podspec]: Let's revert the changes on this file.

File: buildSrc/src/main/java/Configuration.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+object Modules {
+    const val SHARED = ":shared"
+}
+
+object Plugins {
+    const val ANDROID = "android"
+    const val ANDROID_APPLICATION = "com.android.application"
+    const val ANDROID_LIBRARY = "com.android.library"
+
+    const val COCOAPODS = "native.cocoapods"
```
Comment (markgravity) [inline buildSrc/src/main/java/Configuration.kt]: Should we use `IOS_COCOAPODS` as we are using `ANDROID_xxx` above?🙏

Comment (kaungkhantsoe) [reply buildSrc/src/main/java/Configuration.kt]: Updated in https://github.com/nimblehq/kmm-templates/pull/25/commits/28f52e9f0fccca5bb530e2c905b90b030c46112c

**Review by markgravity — APPROVED** [review]

**Review by phongvhd93 — APPROVED** [review]

File: buildSrc/src/main/java/Dependencies.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const val ANDROID_MIN_SDK_VERSION = 24
+    const val ANDROID_TARGET_SDK_VERSION = 33
+
+    const val ANDROID_VERSION_CODE = 1
+    const val ANDROID_VERSION_NAME = "1.0.0"
+
+    const val ANDROID_APPLICATION = "8.0.2"
+    const val ANDROID_LIBRARY = "8.0.2"
+    const val ANDROID = "1.8.21"
+    const val MULTIPLATFORM = "1.8.21"
```
Comment (suho) [inline buildSrc/src/main/java/Dependencies.kt]: This is kotlin version, not ANDROID or MULTIPLATFORM

File: shared/src/androidMain/kotlin/co/nimblehq/kmm/template/di/module/PlatformModule.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package co.nimblehq.kmm.template.di.module
+
+import org.koin.core.module.Module
+import org.koin.dsl.module
+
+actual fun platformModule(): Module = module {
+
```
Comment (suho) [inline shared/src/androidMain/kotlin/co/nimblehq/kmm/template/di/module/PlatformModule.kt:7]: Please add TODO here

Comment (kaungkhantsoe) [reply shared/src/androidMain/kotlin/co/nimblehq/kmm/template/di/module/PlatformModule.kt:7]: Updated in https://github.com/nimblehq/kmm-templates/pull/25/commits/fdd5696df9c927e082a25dfc5261a991eb8d88a6
